code/mob/living/carbon/human.dm

```diff
@@ -3185,10 +3185,18 @@
 		// Make copy of item on ground
 		var/obj/item/outChestItem = src.chest_item
 		outChestItem.set_loc(get_turf(src))
-		src.chest_item.AttackSelf(src)
+		activate_chest_item()
 		src.chest_item = null
 		return
+	activate_chest_item()
+
+///Little wrapper proc to clean up the reference in case AttackSelf moves the chest item outside the mob (items that dispose inside the mob get handled in /obj/item/disposing())
+/mob/living/carbon/human/proc/activate_chest_item() //For edge cases like mousetrap roller assemblies
 	src.chest_item.AttackSelf(src)
+	SPAWN(1 SECOND)
```

Comment:
This spawn is sorta ugly; would rather check if something is a chest item in humans' Exited() so if the item leaves for any reason it gets properly removed from the chest.

Reply:
Alternately listen for the loc setting signal (COMSIG_MOVABLE_SET_LOC) on the chest item and call some dummy chest-item-removal proc when that happens (and then unregister from the signal)